src/router/index.ts

```diff
@@ -14,4 +18,37 @@ const router = new VueRouter({
   }
 })
 
+router.beforeEach((to, from, next) => {
+  // If the route requires authentication
+  if (to.matched.some(record => record.meta.requiresAuth)) {
+    // If we have a token
+    if (sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)) {
+      // If there are allowed or disabled roles specified on the route, check if
+      //  the user has those roles, else route to signout
+      if (!KeyCloakService.verifyRoles(to.meta.allowedRoles, to.meta.disabledRoles)) {
```

Comment:
We don't specify `allowedRoles` or `disabledRoles` on our routes. Do we need to? If not then this code can be greatly simplified.

Reply:
Removed - token validation is done through the api, front-end simply checks for its existence.